Index: open-zwave/cpp/src/command_classes/WakeUp.cpp
===================================================================
--- open-zwave.orig/cpp/src/command_classes/WakeUp.cpp	2012-12-19 18:34:12.845548688 +0100
+++ open-zwave/cpp/src/command_classes/WakeUp.cpp	2012-12-19 18:35:06.591141343 +0100
@@ -61,7 +61,7 @@
 ):
 	CommandClass( _homeId, _nodeId ), 
 	m_mutex( new Mutex() ),
-	m_awake( true ),
+	m_awake( false ),
 	m_pollRequired( false ),
 	m_notification( false )
 {
