diff -u -r open-zwave.orig/cpp/src/command_classes/Alarm.cpp open-zwave/cpp/src/command_classes/Alarm.cpp
--- open-zwave.orig/cpp/src/command_classes/Alarm.cpp	2012-12-23 09:54:43.545905000 +0100
+++ open-zwave/cpp/src/command_classes/Alarm.cpp	2012-12-23 09:58:55.546500360 +0100
@@ -88,7 +88,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( AlarmCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp
--- open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-23 09:54:43.533905543 +0100
+++ open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-23 09:58:55.554499998 +0100
@@ -100,7 +100,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( AssociationCommandConfigurationCmd_SupportedRecordsGet );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -122,7 +122,7 @@
 	msg->Append( AssociationCommandConfigurationCmd_Get );
 	msg->Append( _groupIdx );
 	msg->Append( _nodeId );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
@@ -269,7 +269,7 @@
 		msg->Append( _data[i] );
 	}
 
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
diff -u -r open-zwave.orig/cpp/src/command_classes/Association.cpp open-zwave/cpp/src/command_classes/Association.cpp
--- open-zwave.orig/cpp/src/command_classes/Association.cpp	2012-12-23 09:54:43.521906084 +0100
+++ open-zwave/cpp/src/command_classes/Association.cpp	2012-12-23 09:58:55.558499818 +0100
@@ -174,7 +174,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( AssociationCmd_GroupingsGet );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -333,7 +333,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( AssociationCmd_Get );
 	msg->Append( _groupIdx );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
@@ -356,7 +356,7 @@
 	msg->Append( AssociationCmd_Set );
 	msg->Append( _groupIdx );
 	msg->Append( _targetNodeId );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
@@ -379,7 +379,7 @@
 	msg->Append( AssociationCmd_Remove );
 	msg->Append( _groupIdx );
 	msg->Append( _targetNodeId );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
diff -u -r open-zwave.orig/cpp/src/command_classes/Basic.cpp open-zwave/cpp/src/command_classes/Basic.cpp
--- open-zwave.orig/cpp/src/command_classes/Basic.cpp	2012-12-23 09:54:43.505906810 +0100
+++ open-zwave/cpp/src/command_classes/Basic.cpp	2012-12-23 09:58:55.558499818 +0100
@@ -83,7 +83,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( BasicCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -147,7 +147,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( BasicCmd_Set );
 		msg->Append( value->GetValue() );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp
--- open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-23 09:54:43.517906266 +0100
+++ open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-23 09:58:55.562499637 +0100
@@ -78,7 +78,7 @@
 			msg->Append( GetCommandClassId() );
 			msg->Append( BasicWindowCoveringCmd_StartLevelChange );
 			msg->Append( action );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 			return true;
 		}
@@ -91,7 +91,7 @@
 			msg->Append( 2 );
 			msg->Append( GetCommandClassId() );
 			msg->Append( BasicWindowCoveringCmd_StopLevelChange );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 			return true;
 		}
diff -u -r open-zwave.orig/cpp/src/command_classes/Battery.cpp open-zwave/cpp/src/command_classes/Battery.cpp
--- open-zwave.orig/cpp/src/command_classes/Battery.cpp	2012-12-23 09:54:43.553904638 +0100
+++ open-zwave/cpp/src/command_classes/Battery.cpp	2012-12-23 09:58:55.562499637 +0100
@@ -85,7 +85,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( BatteryCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp
--- open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-23 09:54:43.545905000 +0100
+++ open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-23 09:58:55.566499456 +0100
@@ -150,7 +150,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ClimateControlScheduleCmd_ChangedGet );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -241,7 +241,7 @@
 					msg->Append( GetCommandClassId() );
 					msg->Append( ClimateControlScheduleCmd_Get );
 					msg->Append( i );
-					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 					GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 				}
 			}
@@ -254,7 +254,7 @@
 			msg->Append( 2 );
 			msg->Append( GetCommandClassId() );
 			msg->Append( ClimateControlScheduleCmd_OverrideGet );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		}
 		return true;
@@ -351,7 +351,7 @@
 			}
 		}
 
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	}
 	else
@@ -372,7 +372,7 @@
 			msg->Append( ClimateControlScheduleCmd_OverrideSet );
 			msg->Append( (uint8)item.m_value );
 			msg->Append( setback->GetValue() );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		}
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/Clock.cpp open-zwave/cpp/src/command_classes/Clock.cpp
--- open-zwave.orig/cpp/src/command_classes/Clock.cpp	2012-12-23 09:54:43.565904092 +0100
+++ open-zwave/cpp/src/command_classes/Clock.cpp	2012-12-23 09:58:55.570499275 +0100
@@ -101,7 +101,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ClockCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -178,7 +178,7 @@
 		msg->Append( ClockCmd_Set );
 		msg->Append( ( day << 5 ) | hour );
 		msg->Append( minute );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		ret = true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/Configuration.cpp open-zwave/cpp/src/command_classes/Configuration.cpp
--- open-zwave.orig/cpp/src/command_classes/Configuration.cpp	2012-12-23 09:54:43.553904638 +0100
+++ open-zwave/cpp/src/command_classes/Configuration.cpp	2012-12-23 09:58:55.574499094 +0100
@@ -225,7 +225,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( ConfigurationCmd_Get );
 	msg->Append( _parameter );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -260,6 +260,6 @@
 		msg->Append( (uint8)(_value>>8) );
 	}
 	msg->Append( (uint8)_value );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp open-zwave/cpp/src/command_classes/EnergyProduction.cpp
--- open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp	2012-12-23 09:54:43.529905723 +0100
+++ open-zwave/cpp/src/command_classes/EnergyProduction.cpp	2012-12-23 09:58:55.578498913 +0100
@@ -103,7 +103,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( EnergyProductionCmd_Get );
 	msg->Append( _valueEnum );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/Indicator.cpp open-zwave/cpp/src/command_classes/Indicator.cpp
--- open-zwave.orig/cpp/src/command_classes/Indicator.cpp	2012-12-23 09:54:43.521906084 +0100
+++ open-zwave/cpp/src/command_classes/Indicator.cpp	2012-12-23 09:58:55.582498732 +0100
@@ -81,7 +81,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( IndicatorCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -133,7 +133,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( IndicatorCmd_Set );
 		msg->Append( value->GetValue() );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/Language.cpp open-zwave/cpp/src/command_classes/Language.cpp
--- open-zwave.orig/cpp/src/command_classes/Language.cpp	2012-12-23 09:54:43.505906810 +0100
+++ open-zwave/cpp/src/command_classes/Language.cpp	2012-12-23 09:58:55.582498732 +0100
@@ -92,7 +92,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( LanguageCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/Lock.cpp open-zwave/cpp/src/command_classes/Lock.cpp
--- open-zwave.orig/cpp/src/command_classes/Lock.cpp	2012-12-23 09:54:43.509906629 +0100
+++ open-zwave/cpp/src/command_classes/Lock.cpp	2012-12-23 09:58:55.582498732 +0100
@@ -81,7 +81,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( LockCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -133,7 +133,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( LockCmd_Set );
 		msg->Append( value->GetValue() ? 0xff:0x00 );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp
--- open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-23 09:54:43.497907171 +0100
+++ open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-23 09:58:55.586498551 +0100
@@ -95,7 +95,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ManufacturerSpecificCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/Meter.cpp open-zwave/cpp/src/command_classes/Meter.cpp
--- open-zwave.orig/cpp/src/command_classes/Meter.cpp	2012-12-23 09:54:43.553904638 +0100
+++ open-zwave/cpp/src/command_classes/Meter.cpp	2012-12-23 09:58:55.590498370 +0100
@@ -160,7 +160,7 @@
 			msg->Append( 2 );
 			msg->Append( GetCommandClassId() );
 			msg->Append( MeterCmd_SupportedGet );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, _queue );
 			res = true;
 		}
@@ -202,7 +202,7 @@
 			msg->Append( GetCommandClassId() );
 			msg->Append( MeterCmd_Get );
 			msg->Append( (uint8)( i << 3 ) );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, _queue );
 			res |= true;
 		}
@@ -516,7 +516,7 @@
 			msg->Append( 2 );
 			msg->Append( GetCommandClassId() );
 			msg->Append( MeterCmd_Reset );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 			return true;
 		}
diff -u -r open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp open-zwave/cpp/src/command_classes/MeterPulse.cpp
--- open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp	2012-12-23 09:54:43.541905181 +0100
+++ open-zwave/cpp/src/command_classes/MeterPulse.cpp	2012-12-23 09:58:55.590498370 +0100
@@ -80,7 +80,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( MeterPulseCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp open-zwave/cpp/src/command_classes/MultiInstance.cpp
--- open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp	2012-12-23 09:54:43.505906810 +0100
+++ open-zwave/cpp/src/command_classes/MultiInstance.cpp	2012-12-23 09:58:55.598498008 +0100
@@ -198,7 +198,7 @@
 					msg->Append( GetCommandClassId() );
 					msg->Append( MultiInstanceCmd_Get );
 					msg->Append( cc->GetCommandClassId() );
-					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 					GetDriver()->SendMsg( msg, Driver::MsgQueue_Query );
 					res = true;
 				}
@@ -216,7 +216,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( MultiChannelCmd_EndPointGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Query );
 		res = true;
 	}
@@ -384,7 +384,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( MultiChannelCmd_CapabilityGet );
 		msg->Append( i );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	}
 }
@@ -557,7 +557,7 @@
 			msg->Append( GetCommandClassId() );
 			msg->Append( MultiChannelCmd_CapabilityGet );
 			msg->Append( endPoint );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		}
 	}
@@ -583,7 +583,7 @@
 					msg->Append( MultiChannelCmd_EndPointFind );
 					msg->Append( c_genericClass[m_endPointFindIndex] );		// Generic device class
 					msg->Append( 0xff );									// Any specific device class
-					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+					msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 					GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 				}
 			}
diff -u -r open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp open-zwave/cpp/src/command_classes/NodeNaming.cpp
--- open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp	2012-12-23 09:54:43.557904456 +0100
+++ open-zwave/cpp/src/command_classes/NodeNaming.cpp	2012-12-23 09:58:55.602497827 +0100
@@ -253,7 +253,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( NodeNamingCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -266,7 +266,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( NodeNamingCmd_LocationGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -349,7 +349,7 @@
 		msg->Append( _name[i] );
 	}
 
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
@@ -381,7 +381,7 @@
 		msg->Append( _location[i] );
 	}
 
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
diff -u -r open-zwave.orig/cpp/src/command_classes/NoOperation.cpp open-zwave/cpp/src/command_classes/NoOperation.cpp
--- open-zwave.orig/cpp/src/command_classes/NoOperation.cpp	2012-12-23 09:54:43.549904819 +0100
+++ open-zwave/cpp/src/command_classes/NoOperation.cpp	2012-12-23 09:58:55.602497827 +0100
@@ -68,7 +68,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( 0 );
 	if( _route )
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	else
 		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_NO_ROUTE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_NoOp );
diff -u -r open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp open-zwave/cpp/src/command_classes/Powerlevel.cpp
--- open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp	2012-12-23 09:54:43.525905903 +0100
+++ open-zwave/cpp/src/command_classes/Powerlevel.cpp	2012-12-23 09:58:55.606497646 +0100
@@ -126,7 +126,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( PowerlevelCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -341,7 +341,7 @@
 	msg->Append( PowerlevelCmd_Set );
 	msg->Append( (uint8)powerLevel );
 	msg->Append( timeout );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -401,7 +401,7 @@
 	msg->Append( (uint8)powerLevel );
 	msg->Append( (uint8)(numFrames >> 8) );
 	msg->Append( (uint8)(numFrames & 0x00ff) );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -422,7 +422,7 @@
 	msg->Append( 6 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( PowerlevelCmd_TestNodeGet );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/Protection.cpp open-zwave/cpp/src/command_classes/Protection.cpp
--- open-zwave.orig/cpp/src/command_classes/Protection.cpp	2012-12-23 09:54:43.561904274 +0100
+++ open-zwave/cpp/src/command_classes/Protection.cpp	2012-12-23 09:58:55.606497646 +0100
@@ -90,7 +90,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ProtectionCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -143,7 +143,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( ProtectionCmd_Set );
 		msg->Append( (uint8)item.m_value );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp open-zwave/cpp/src/command_classes/SensorAlarm.cpp
--- open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp	2012-12-23 09:54:43.545905000 +0100
+++ open-zwave/cpp/src/command_classes/SensorAlarm.cpp	2012-12-23 09:58:55.610497465 +0100
@@ -125,7 +125,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( SensorAlarmCmd_SupportedGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 	}
 	else
@@ -138,7 +138,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( SensorAlarmCmd_Get );
 		msg->Append( _alarmType );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 	}
 	return true;
diff -u -r open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp open-zwave/cpp/src/command_classes/SensorBinary.cpp
--- open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp	2012-12-23 09:54:43.501906991 +0100
+++ open-zwave/cpp/src/command_classes/SensorBinary.cpp	2012-12-23 09:58:55.610497465 +0100
@@ -80,7 +80,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SensorBinaryCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp open-zwave/cpp/src/command_classes/SensorMultilevel.cpp
--- open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-23 09:54:43.501906991 +0100
+++ open-zwave/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-23 09:58:55.610497465 +0100
@@ -145,7 +145,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SensorMultilevelCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp open-zwave/cpp/src/command_classes/SwitchAll.cpp
--- open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp	2012-12-23 09:54:43.509906629 +0100
+++ open-zwave/cpp/src/command_classes/SwitchAll.cpp	2012-12-23 09:58:55.614497284 +0100
@@ -91,7 +91,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchAllCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -143,7 +143,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( SwitchAllCmd_Set );
 		msg->Append( (uint8)item.m_value );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
@@ -167,7 +167,7 @@
 	msg->Append( 2 );
 	msg->Append( StaticGetCommandClassId() );
 	msg->Append( SwitchAllCmd_Off );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	_driver->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
@@ -187,7 +187,7 @@
 	msg->Append( 2 );
 	msg->Append( StaticGetCommandClassId() );
 	msg->Append( SwitchAllCmd_On );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	_driver->SendMsg( msg, Driver::MsgQueue_Send );
 }
 
diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp open-zwave/cpp/src/command_classes/SwitchBinary.cpp
--- open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp	2012-12-23 09:54:43.553904638 +0100
+++ open-zwave/cpp/src/command_classes/SwitchBinary.cpp	2012-12-23 09:58:55.614497284 +0100
@@ -81,7 +81,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchBinaryCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -133,7 +133,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( SwitchBinaryCmd_Set );
 		msg->Append( value->GetValue() ? 0xff : 0x00 );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp
--- open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-23 09:54:43.549904819 +0100
+++ open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-23 09:58:55.618497103 +0100
@@ -142,7 +142,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( SwitchMultilevelCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -235,7 +235,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( SwitchMultilevelCmd_SupportedGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 
 		// Set the request flag again - it will be cleared when we get a
@@ -427,7 +427,7 @@
 		msg->Append( _level );
 	}
 
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -507,7 +507,7 @@
 		msg->Append( step );
 	}
 
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -528,7 +528,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchMultilevelCmd_StopLevelChange );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp
--- open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-23 09:54:43.517906266 +0100
+++ open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-23 09:58:55.618497103 +0100
@@ -81,7 +81,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleBinaryCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -128,7 +128,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleBinaryCmd_Set );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp
--- open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-23 09:54:43.517906266 +0100
+++ open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-23 09:58:55.618497103 +0100
@@ -84,7 +84,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleMultilevelCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -131,7 +131,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleMultilevelCmd_Set );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 	return true;
 }
@@ -158,7 +158,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleMultilevelCmd_StartLevelChange );
 	msg->Append( param );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 }
 
 //-----------------------------------------------------------------------------
@@ -175,7 +175,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( SwitchToggleMultilevelCmd_StopLevelChange );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 }
 
 //-----------------------------------------------------------------------------
diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp
--- open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-23 09:54:43.541905181 +0100
+++ open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-23 09:58:55.622496922 +0100
@@ -186,7 +186,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatFanModeCmd_SupportedGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -200,7 +200,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatFanModeCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -302,7 +302,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatFanModeCmd_Set );
 		msg->Append( state );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp open-zwave/cpp/src/command_classes/ThermostatFanState.cpp
--- open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-23 09:54:43.557904456 +0100
+++ open-zwave/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-23 09:58:55.622496922 +0100
@@ -101,7 +101,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ThermostatFanStateCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp open-zwave/cpp/src/command_classes/ThermostatMode.cpp
--- open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp	2012-12-23 09:54:43.493907352 +0100
+++ open-zwave/cpp/src/command_classes/ThermostatMode.cpp	2012-12-23 09:58:55.626496741 +0100
@@ -197,7 +197,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatModeCmd_SupportedGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -211,7 +211,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatModeCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -314,7 +314,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatModeCmd_Set );
 		msg->Append( state );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp
--- open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-23 09:54:43.513906448 +0100
+++ open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-23 09:58:55.626496741 +0100
@@ -99,7 +99,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( ThermostatOperatingStateCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp
--- open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-23 09:54:43.521906084 +0100
+++ open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-23 09:58:55.626496741 +0100
@@ -180,7 +180,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatSetpointCmd_SupportedGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -197,7 +197,7 @@
 		msg->Append( GetCommandClassId() );
 		msg->Append( ThermostatSetpointCmd_Get );
 		msg->Append( _setPointIndex );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -292,7 +292,7 @@
 		msg->Append( ThermostatSetpointCmd_Set );
 		msg->Append( value->GetID().GetIndex() );
 		AppendValue( msg, value->GetValue(), scale );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/UserCode.cpp open-zwave/cpp/src/command_classes/UserCode.cpp
--- open-zwave.orig/cpp/src/command_classes/UserCode.cpp	2012-12-23 09:54:43.517906266 +0100
+++ open-zwave/cpp/src/command_classes/UserCode.cpp	2012-12-23 09:58:55.630496560 +0100
@@ -142,7 +142,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( UserNumberCmd_Get );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 		return true;
 	}
@@ -153,7 +153,7 @@
 	msg->Append( GetCommandClassId() );
 	msg->Append( UserCodeCmd_Get );
 	msg->Append( _userCodeIdx );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -268,7 +268,7 @@
 		{
 			msg->Append( s[i] );
 		}
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 		return true;
 	}
diff -u -r open-zwave.orig/cpp/src/command_classes/Version.cpp open-zwave/cpp/src/command_classes/Version.cpp
--- open-zwave.orig/cpp/src/command_classes/Version.cpp	2012-12-23 09:54:43.525905903 +0100
+++ open-zwave/cpp/src/command_classes/Version.cpp	2012-12-23 09:58:55.630496560 +0100
@@ -144,7 +144,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( VersionCmd_Get );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 	return true;
 }
@@ -229,7 +229,7 @@
 			msg->Append( GetCommandClassId() );
 			msg->Append( VersionCmd_CommandClassGet );
 			msg->Append( _commandClass->GetCommandClassId() );
-			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+			msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
 			return true;
 		}
diff -u -r open-zwave.orig/cpp/src/command_classes/WakeUp.cpp open-zwave/cpp/src/command_classes/WakeUp.cpp
--- open-zwave.orig/cpp/src/command_classes/WakeUp.cpp	2012-12-23 09:54:43.513906448 +0100
+++ open-zwave/cpp/src/command_classes/WakeUp.cpp	2012-12-23 09:58:55.634496379 +0100
@@ -152,7 +152,7 @@
 	msg->Append( 2 );
 	msg->Append( GetCommandClassId() );
 	msg->Append( WakeUpCmd_IntervalGet );
-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 	GetDriver()->SendMsg( msg, _queue );
 
 	if( GetVersion() > 1 )
@@ -162,7 +162,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( WakeUpCmd_IntervalCapabilitiesGet );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, _queue );
 	}
 
@@ -286,7 +286,7 @@
 		msg->Append( (uint8)(( interval >> 8 ) & 0xff) );	 
 		msg->Append( (uint8)( interval & 0xff ) );		
 		msg->Append( GetDriver()->GetNodeId() );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_WakeUp );
 		return true;
 	}
@@ -423,7 +423,7 @@
 		msg->Append( 2 );
 		msg->Append( GetCommandClassId() );
 		msg->Append( WakeUpCmd_NoMoreInformation );
-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
+		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
 		GetDriver()->SendMsg( msg, Driver::MsgQueue_WakeUp );
 	}
 }
diff -u -r open-zwave.orig/debian/patches/explore_routing.patch open-zwave/debian/patches/explore_routing.patch
--- open-zwave.orig/debian/patches/explore_routing.patch	2012-12-23 09:54:44.661854477 +0100
+++ open-zwave/debian/patches/explore_routing.patch	2012-12-23 09:59:26.205113369 +0100
@@ -1,6 +1,6 @@
-diff -ru open-zwave.orig/cpp/src/command_classes/Alarm.cpp open-zwave/cpp/src/command_classes/Alarm.cpp
---- open-zwave.orig/cpp/src/command_classes/Alarm.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/Alarm.cpp	2012-12-19 17:11:37.405635661 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Alarm.cpp open-zwave/cpp/src/command_classes/Alarm.cpp
+--- open-zwave.orig/cpp/src/command_classes/Alarm.cpp	2012-12-23 09:54:43.545905000 +0100
++++ open-zwave/cpp/src/command_classes/Alarm.cpp	2012-12-23 09:58:55.546500360 +0100
 @@ -88,7 +88,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -10,9 +10,9 @@
  		GetDriver()->SendMsg( msg, _queue );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp
---- open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-19 17:11:37.409635477 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp
+--- open-zwave.orig/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-23 09:54:43.533905543 +0100
++++ open-zwave/cpp/src/command_classes/AssociationCommandConfiguration.cpp	2012-12-23 09:58:55.554499998 +0100
 @@ -100,7 +100,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -40,9 +40,9 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  }
  
-diff -ru open-zwave.orig/cpp/src/command_classes/Association.cpp open-zwave/cpp/src/command_classes/Association.cpp
---- open-zwave.orig/cpp/src/command_classes/Association.cpp	2012-12-19 17:09:52.194453252 +0100
-+++ open-zwave/cpp/src/command_classes/Association.cpp	2012-12-19 17:11:37.413635294 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Association.cpp open-zwave/cpp/src/command_classes/Association.cpp
+--- open-zwave.orig/cpp/src/command_classes/Association.cpp	2012-12-23 09:54:43.521906084 +0100
++++ open-zwave/cpp/src/command_classes/Association.cpp	2012-12-23 09:58:55.558499818 +0100
 @@ -174,7 +174,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -79,10 +79,10 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  }
  
-diff -ru open-zwave.orig/cpp/src/command_classes/Basic.cpp open-zwave/cpp/src/command_classes/Basic.cpp
---- open-zwave.orig/cpp/src/command_classes/Basic.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/Basic.cpp	2012-12-19 17:11:37.413635294 +0100
-@@ -82,7 +82,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/Basic.cpp open-zwave/cpp/src/command_classes/Basic.cpp
+--- open-zwave.orig/cpp/src/command_classes/Basic.cpp	2012-12-23 09:54:43.505906810 +0100
++++ open-zwave/cpp/src/command_classes/Basic.cpp	2012-12-23 09:58:55.558499818 +0100
+@@ -83,7 +83,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
  	msg->Append( BasicCmd_Get );
@@ -91,7 +91,7 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-@@ -146,7 +146,7 @@
+@@ -147,7 +147,7 @@
  		msg->Append( GetCommandClassId() );
  		msg->Append( BasicCmd_Set );
  		msg->Append( value->GetValue() );
@@ -100,9 +100,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp
---- open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-19 17:09:52.166454534 +0100
-+++ open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-19 17:11:37.413635294 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp
+--- open-zwave.orig/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-23 09:54:43.517906266 +0100
++++ open-zwave/cpp/src/command_classes/BasicWindowCovering.cpp	2012-12-23 09:58:55.562499637 +0100
 @@ -78,7 +78,7 @@
  			msg->Append( GetCommandClassId() );
  			msg->Append( BasicWindowCoveringCmd_StartLevelChange );
@@ -121,9 +121,9 @@
  			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  			return true;
  		}
-diff -ru open-zwave.orig/cpp/src/command_classes/Battery.cpp open-zwave/cpp/src/command_classes/Battery.cpp
---- open-zwave.orig/cpp/src/command_classes/Battery.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/Battery.cpp	2012-12-19 17:11:37.413635294 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Battery.cpp open-zwave/cpp/src/command_classes/Battery.cpp
+--- open-zwave.orig/cpp/src/command_classes/Battery.cpp	2012-12-23 09:54:43.553904638 +0100
++++ open-zwave/cpp/src/command_classes/Battery.cpp	2012-12-23 09:58:55.562499637 +0100
 @@ -85,7 +85,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -133,9 +133,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp
---- open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-19 17:09:52.198453069 +0100
-+++ open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-19 17:11:37.417635111 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp
+--- open-zwave.orig/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-23 09:54:43.545905000 +0100
++++ open-zwave/cpp/src/command_classes/ClimateControlSchedule.cpp	2012-12-23 09:58:55.566499456 +0100
 @@ -150,7 +150,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -181,9 +181,9 @@
  			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		}
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/Clock.cpp open-zwave/cpp/src/command_classes/Clock.cpp
---- open-zwave.orig/cpp/src/command_classes/Clock.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/Clock.cpp	2012-12-19 17:11:37.417635111 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Clock.cpp open-zwave/cpp/src/command_classes/Clock.cpp
+--- open-zwave.orig/cpp/src/command_classes/Clock.cpp	2012-12-23 09:54:43.565904092 +0100
++++ open-zwave/cpp/src/command_classes/Clock.cpp	2012-12-23 09:58:55.570499275 +0100
 @@ -101,7 +101,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -202,10 +202,10 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		ret = true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/Configuration.cpp open-zwave/cpp/src/command_classes/Configuration.cpp
---- open-zwave.orig/cpp/src/command_classes/Configuration.cpp	2012-12-19 17:09:52.190453435 +0100
-+++ open-zwave/cpp/src/command_classes/Configuration.cpp	2012-12-19 17:11:37.425634745 +0100
-@@ -218,7 +218,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/Configuration.cpp open-zwave/cpp/src/command_classes/Configuration.cpp
+--- open-zwave.orig/cpp/src/command_classes/Configuration.cpp	2012-12-23 09:54:43.553904638 +0100
++++ open-zwave/cpp/src/command_classes/Configuration.cpp	2012-12-23 09:58:55.574499094 +0100
+@@ -225,7 +225,7 @@
  	msg->Append( GetCommandClassId() );
  	msg->Append( ConfigurationCmd_Get );
  	msg->Append( _parameter );
@@ -214,7 +214,7 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	return true;
  }
-@@ -253,6 +253,6 @@
+@@ -260,6 +260,6 @@
  		msg->Append( (uint8)(_value>>8) );
  	}
  	msg->Append( (uint8)_value );
@@ -222,9 +222,9 @@
 +	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp open-zwave/cpp/src/command_classes/EnergyProduction.cpp
---- open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp	2012-12-19 17:09:52.170454351 +0100
-+++ open-zwave/cpp/src/command_classes/EnergyProduction.cpp	2012-12-19 17:11:37.425634745 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp open-zwave/cpp/src/command_classes/EnergyProduction.cpp
+--- open-zwave.orig/cpp/src/command_classes/EnergyProduction.cpp	2012-12-23 09:54:43.529905723 +0100
++++ open-zwave/cpp/src/command_classes/EnergyProduction.cpp	2012-12-23 09:58:55.578498913 +0100
 @@ -103,7 +103,7 @@
  	msg->Append( GetCommandClassId() );
  	msg->Append( EnergyProductionCmd_Get );
@@ -234,9 +234,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/Indicator.cpp open-zwave/cpp/src/command_classes/Indicator.cpp
---- open-zwave.orig/cpp/src/command_classes/Indicator.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/Indicator.cpp	2012-12-19 17:11:37.429634562 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Indicator.cpp open-zwave/cpp/src/command_classes/Indicator.cpp
+--- open-zwave.orig/cpp/src/command_classes/Indicator.cpp	2012-12-23 09:54:43.521906084 +0100
++++ open-zwave/cpp/src/command_classes/Indicator.cpp	2012-12-23 09:58:55.582498732 +0100
 @@ -81,7 +81,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -255,9 +255,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/Language.cpp open-zwave/cpp/src/command_classes/Language.cpp
---- open-zwave.orig/cpp/src/command_classes/Language.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/Language.cpp	2012-12-19 17:11:37.429634562 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Language.cpp open-zwave/cpp/src/command_classes/Language.cpp
+--- open-zwave.orig/cpp/src/command_classes/Language.cpp	2012-12-23 09:54:43.505906810 +0100
++++ open-zwave/cpp/src/command_classes/Language.cpp	2012-12-23 09:58:55.582498732 +0100
 @@ -92,7 +92,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -267,9 +267,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/Lock.cpp open-zwave/cpp/src/command_classes/Lock.cpp
---- open-zwave.orig/cpp/src/command_classes/Lock.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/Lock.cpp	2012-12-19 17:11:37.429634562 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Lock.cpp open-zwave/cpp/src/command_classes/Lock.cpp
+--- open-zwave.orig/cpp/src/command_classes/Lock.cpp	2012-12-23 09:54:43.509906629 +0100
++++ open-zwave/cpp/src/command_classes/Lock.cpp	2012-12-23 09:58:55.582498732 +0100
 @@ -81,7 +81,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -288,9 +288,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp
---- open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-19 17:09:52.166454534 +0100
-+++ open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-19 17:11:37.433634379 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp
+--- open-zwave.orig/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-23 09:54:43.497907171 +0100
++++ open-zwave/cpp/src/command_classes/ManufacturerSpecific.cpp	2012-12-23 09:58:55.586498551 +0100
 @@ -95,7 +95,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -300,9 +300,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/Meter.cpp open-zwave/cpp/src/command_classes/Meter.cpp
---- open-zwave.orig/cpp/src/command_classes/Meter.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/Meter.cpp	2012-12-19 17:11:37.437634196 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/Meter.cpp open-zwave/cpp/src/command_classes/Meter.cpp
+--- open-zwave.orig/cpp/src/command_classes/Meter.cpp	2012-12-23 09:54:43.553904638 +0100
++++ open-zwave/cpp/src/command_classes/Meter.cpp	2012-12-23 09:58:55.590498370 +0100
 @@ -160,7 +160,7 @@
  			msg->Append( 2 );
  			msg->Append( GetCommandClassId() );
@@ -330,9 +330,9 @@
  			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  			return true;
  		}
-diff -ru open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp open-zwave/cpp/src/command_classes/MeterPulse.cpp
---- open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp	2012-12-19 17:09:52.314447756 +0100
-+++ open-zwave/cpp/src/command_classes/MeterPulse.cpp	2012-12-19 17:11:37.437634196 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp open-zwave/cpp/src/command_classes/MeterPulse.cpp
+--- open-zwave.orig/cpp/src/command_classes/MeterPulse.cpp	2012-12-23 09:54:43.541905181 +0100
++++ open-zwave/cpp/src/command_classes/MeterPulse.cpp	2012-12-23 09:58:55.590498370 +0100
 @@ -80,7 +80,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -342,10 +342,10 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp open-zwave/cpp/src/command_classes/MultiInstance.cpp
---- open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/MultiInstance.cpp	2012-12-19 17:11:37.441634013 +0100
-@@ -193,7 +193,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp open-zwave/cpp/src/command_classes/MultiInstance.cpp
+--- open-zwave.orig/cpp/src/command_classes/MultiInstance.cpp	2012-12-23 09:54:43.505906810 +0100
++++ open-zwave/cpp/src/command_classes/MultiInstance.cpp	2012-12-23 09:58:55.598498008 +0100
+@@ -198,7 +198,7 @@
  					msg->Append( GetCommandClassId() );
  					msg->Append( MultiInstanceCmd_Get );
  					msg->Append( cc->GetCommandClassId() );
@@ -354,7 +354,7 @@
  					GetDriver()->SendMsg( msg, Driver::MsgQueue_Query );
  					res = true;
  				}
-@@ -211,7 +211,7 @@
+@@ -216,7 +216,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
  		msg->Append( MultiChannelCmd_EndPointGet );
@@ -363,7 +363,7 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Query );
  		res = true;
  	}
-@@ -383,7 +383,7 @@
+@@ -384,7 +384,7 @@
  		msg->Append( GetCommandClassId() );
  		msg->Append( MultiChannelCmd_CapabilityGet );
  		msg->Append( i );
@@ -372,7 +372,7 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	}
  }
-@@ -556,7 +556,7 @@
+@@ -557,7 +557,7 @@
  			msg->Append( GetCommandClassId() );
  			msg->Append( MultiChannelCmd_CapabilityGet );
  			msg->Append( endPoint );
@@ -381,7 +381,7 @@
  			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		}
  	}
-@@ -582,7 +582,7 @@
+@@ -583,7 +583,7 @@
  					msg->Append( MultiChannelCmd_EndPointFind );
  					msg->Append( c_genericClass[m_endPointFindIndex] );		// Generic device class
  					msg->Append( 0xff );									// Any specific device class
@@ -390,9 +390,9 @@
  					GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  				}
  			}
-diff -ru open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp open-zwave/cpp/src/command_classes/NodeNaming.cpp
---- open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/NodeNaming.cpp	2012-12-19 17:11:37.445633829 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp open-zwave/cpp/src/command_classes/NodeNaming.cpp
+--- open-zwave.orig/cpp/src/command_classes/NodeNaming.cpp	2012-12-23 09:54:43.557904456 +0100
++++ open-zwave/cpp/src/command_classes/NodeNaming.cpp	2012-12-23 09:58:55.602497827 +0100
 @@ -253,7 +253,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -429,9 +429,9 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  }
  
-diff -ru open-zwave.orig/cpp/src/command_classes/NoOperation.cpp open-zwave/cpp/src/command_classes/NoOperation.cpp
---- open-zwave.orig/cpp/src/command_classes/NoOperation.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/NoOperation.cpp	2012-12-19 17:11:37.445633829 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/NoOperation.cpp open-zwave/cpp/src/command_classes/NoOperation.cpp
+--- open-zwave.orig/cpp/src/command_classes/NoOperation.cpp	2012-12-23 09:54:43.549904819 +0100
++++ open-zwave/cpp/src/command_classes/NoOperation.cpp	2012-12-23 09:58:55.602497827 +0100
 @@ -68,7 +68,7 @@
  	msg->Append( GetCommandClassId() );
  	msg->Append( 0 );
@@ -440,31 +440,49 @@
 +		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
  	else
  		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_NO_ROUTE );
- 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
-diff -ru open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp open-zwave/cpp/src/command_classes/Powerlevel.cpp
---- open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/Powerlevel.cpp	2012-12-19 17:11:37.445633829 +0100
-@@ -121,7 +121,7 @@
+ 	GetDriver()->SendMsg( msg, Driver::MsgQueue_NoOp );
+diff -u -r open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp open-zwave/cpp/src/command_classes/Powerlevel.cpp
+--- open-zwave.orig/cpp/src/command_classes/Powerlevel.cpp	2012-12-23 09:54:43.525905903 +0100
++++ open-zwave/cpp/src/command_classes/Powerlevel.cpp	2012-12-23 09:58:55.606497646 +0100
+@@ -126,7 +126,7 @@
+ 		msg->Append( 2 );
+ 		msg->Append( GetCommandClassId() );
+ 		msg->Append( PowerlevelCmd_Get );
+-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
++		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
+ 		GetDriver()->SendMsg( msg, _queue );
+ 		return true;
+ 	}
+@@ -341,7 +341,7 @@
  	msg->Append( PowerlevelCmd_Set );
- 	msg->Append( (uint8)_powerLevel );
- 	msg->Append( _timeout );
+ 	msg->Append( (uint8)powerLevel );
+ 	msg->Append( timeout );
 -	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
 +	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
+ 	return true;
  }
- 
-@@ -151,7 +151,7 @@
- 	msg->Append( (uint8)_powerLevel );
- 	msg->Append( (uint8)(_numFrames >> 8) );
- 	msg->Append( (uint8)(_numFrames & 0x00ff) );
+@@ -401,7 +401,7 @@
+ 	msg->Append( (uint8)powerLevel );
+ 	msg->Append( (uint8)(numFrames >> 8) );
+ 	msg->Append( (uint8)(numFrames & 0x00ff) );
 -	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
 +	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
+ 	return true;
  }
- 
-diff -ru open-zwave.orig/cpp/src/command_classes/Protection.cpp open-zwave/cpp/src/command_classes/Protection.cpp
---- open-zwave.orig/cpp/src/command_classes/Protection.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/Protection.cpp	2012-12-19 17:11:37.445633829 +0100
+@@ -422,7 +422,7 @@
+ 	msg->Append( 6 );
+ 	msg->Append( GetCommandClassId() );
+ 	msg->Append( PowerlevelCmd_TestNodeGet );
+-	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
++	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
+ 	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
+ 	return true;
+ }
+diff -u -r open-zwave.orig/cpp/src/command_classes/Protection.cpp open-zwave/cpp/src/command_classes/Protection.cpp
+--- open-zwave.orig/cpp/src/command_classes/Protection.cpp	2012-12-23 09:54:43.561904274 +0100
++++ open-zwave/cpp/src/command_classes/Protection.cpp	2012-12-23 09:58:55.606497646 +0100
 @@ -90,7 +90,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -483,9 +501,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp open-zwave/cpp/src/command_classes/SensorAlarm.cpp
---- open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/SensorAlarm.cpp	2012-12-19 17:11:37.449633646 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp open-zwave/cpp/src/command_classes/SensorAlarm.cpp
+--- open-zwave.orig/cpp/src/command_classes/SensorAlarm.cpp	2012-12-23 09:54:43.545905000 +0100
++++ open-zwave/cpp/src/command_classes/SensorAlarm.cpp	2012-12-23 09:58:55.610497465 +0100
 @@ -125,7 +125,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -504,9 +522,9 @@
  		GetDriver()->SendMsg( msg, _queue );
  	}
  	return true;
-diff -ru open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp open-zwave/cpp/src/command_classes/SensorBinary.cpp
---- open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp	2012-12-19 17:09:52.170454351 +0100
-+++ open-zwave/cpp/src/command_classes/SensorBinary.cpp	2012-12-19 17:11:37.449633646 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp open-zwave/cpp/src/command_classes/SensorBinary.cpp
+--- open-zwave.orig/cpp/src/command_classes/SensorBinary.cpp	2012-12-23 09:54:43.501906991 +0100
++++ open-zwave/cpp/src/command_classes/SensorBinary.cpp	2012-12-23 09:58:55.610497465 +0100
 @@ -80,7 +80,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -516,9 +534,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp open-zwave/cpp/src/command_classes/SensorMultilevel.cpp
---- open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-19 17:11:37.449633646 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp open-zwave/cpp/src/command_classes/SensorMultilevel.cpp
+--- open-zwave.orig/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-23 09:54:43.501906991 +0100
++++ open-zwave/cpp/src/command_classes/SensorMultilevel.cpp	2012-12-23 09:58:55.610497465 +0100
 @@ -145,7 +145,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -528,9 +546,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp open-zwave/cpp/src/command_classes/SwitchAll.cpp
---- open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp	2012-12-19 17:09:52.190453435 +0100
-+++ open-zwave/cpp/src/command_classes/SwitchAll.cpp	2012-12-19 17:11:37.453633463 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp open-zwave/cpp/src/command_classes/SwitchAll.cpp
+--- open-zwave.orig/cpp/src/command_classes/SwitchAll.cpp	2012-12-23 09:54:43.509906629 +0100
++++ open-zwave/cpp/src/command_classes/SwitchAll.cpp	2012-12-23 09:58:55.614497284 +0100
 @@ -91,7 +91,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -567,9 +585,9 @@
  	_driver->SendMsg( msg, Driver::MsgQueue_Send );
  }
  
-diff -ru open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp open-zwave/cpp/src/command_classes/SwitchBinary.cpp
---- open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp	2012-12-19 17:09:52.322447390 +0100
-+++ open-zwave/cpp/src/command_classes/SwitchBinary.cpp	2012-12-19 17:11:37.453633463 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp open-zwave/cpp/src/command_classes/SwitchBinary.cpp
+--- open-zwave.orig/cpp/src/command_classes/SwitchBinary.cpp	2012-12-23 09:54:43.553904638 +0100
++++ open-zwave/cpp/src/command_classes/SwitchBinary.cpp	2012-12-23 09:58:55.614497284 +0100
 @@ -81,7 +81,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -588,19 +606,19 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp
---- open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-19 17:09:52.190453435 +0100
-+++ open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-19 17:11:37.457633280 +0100
-@@ -140,7 +140,7 @@
- 	msg->Append( 2 );
- 	msg->Append( GetCommandClassId() );
- 	msg->Append( SwitchMultilevelCmd_Get );
--	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
-+	msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
- 	GetDriver()->SendMsg( msg, _queue );
- 	return true;
- }
-@@ -231,7 +231,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp
+--- open-zwave.orig/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-23 09:54:43.549904819 +0100
++++ open-zwave/cpp/src/command_classes/SwitchMultilevel.cpp	2012-12-23 09:58:55.618497103 +0100
+@@ -142,7 +142,7 @@
+ 		msg->Append( 2 );
+ 		msg->Append( GetCommandClassId() );
+ 		msg->Append( SwitchMultilevelCmd_Get );
+-		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
++		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
+ 		GetDriver()->SendMsg( msg, _queue );
+ 		return true;
+ 	}
+@@ -235,7 +235,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
  		msg->Append( SwitchMultilevelCmd_SupportedGet );
@@ -609,7 +627,7 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  
  		// Set the request flag again - it will be cleared when we get a
-@@ -411,7 +411,7 @@
+@@ -427,7 +427,7 @@
  		msg->Append( _level );
  	}
  
@@ -618,7 +636,7 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	return true;
  }
-@@ -491,7 +491,7 @@
+@@ -507,7 +507,7 @@
  		msg->Append( step );
  	}
  
@@ -627,7 +645,7 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	return true;
  }
-@@ -512,7 +512,7 @@
+@@ -528,7 +528,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
  	msg->Append( SwitchMultilevelCmd_StopLevelChange );
@@ -636,9 +654,9 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp
---- open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-19 17:11:37.457633280 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp
+--- open-zwave.orig/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-23 09:54:43.517906266 +0100
++++ open-zwave/cpp/src/command_classes/SwitchToggleBinary.cpp	2012-12-23 09:58:55.618497103 +0100
 @@ -81,7 +81,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -657,9 +675,9 @@
  	GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp
---- open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-19 17:09:52.318447573 +0100
-+++ open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-19 17:11:37.457633280 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp
+--- open-zwave.orig/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-23 09:54:43.517906266 +0100
++++ open-zwave/cpp/src/command_classes/SwitchToggleMultilevel.cpp	2012-12-23 09:58:55.618497103 +0100
 @@ -84,7 +84,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -696,9 +714,9 @@
  }
  
  //-----------------------------------------------------------------------------
-diff -ru open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp
---- open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-19 17:09:52.174454168 +0100
-+++ open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-19 17:11:37.461633097 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp
+--- open-zwave.orig/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-23 09:54:43.541905181 +0100
++++ open-zwave/cpp/src/command_classes/ThermostatFanMode.cpp	2012-12-23 09:58:55.622496922 +0100
 @@ -186,7 +186,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -726,9 +744,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp open-zwave/cpp/src/command_classes/ThermostatFanState.cpp
---- open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-19 17:09:52.166454534 +0100
-+++ open-zwave/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-19 17:11:37.461633097 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp open-zwave/cpp/src/command_classes/ThermostatFanState.cpp
+--- open-zwave.orig/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-23 09:54:43.557904456 +0100
++++ open-zwave/cpp/src/command_classes/ThermostatFanState.cpp	2012-12-23 09:58:55.622496922 +0100
 @@ -101,7 +101,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -738,9 +756,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp open-zwave/cpp/src/command_classes/ThermostatMode.cpp
---- open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp	2012-12-19 17:09:52.182453801 +0100
-+++ open-zwave/cpp/src/command_classes/ThermostatMode.cpp	2012-12-19 17:11:37.461633097 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp open-zwave/cpp/src/command_classes/ThermostatMode.cpp
+--- open-zwave.orig/cpp/src/command_classes/ThermostatMode.cpp	2012-12-23 09:54:43.493907352 +0100
++++ open-zwave/cpp/src/command_classes/ThermostatMode.cpp	2012-12-23 09:58:55.626496741 +0100
 @@ -197,7 +197,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -768,9 +786,9 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp
---- open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-19 17:09:52.190453435 +0100
-+++ open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-19 17:11:37.465632914 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp
+--- open-zwave.orig/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-23 09:54:43.513906448 +0100
++++ open-zwave/cpp/src/command_classes/ThermostatOperatingState.cpp	2012-12-23 09:58:55.626496741 +0100
 @@ -99,7 +99,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
@@ -780,9 +798,9 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-diff -ru open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp
---- open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-19 17:09:52.166454534 +0100
-+++ open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-19 17:11:37.465632914 +0100
+diff -u -r open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp
+--- open-zwave.orig/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-23 09:54:43.521906084 +0100
++++ open-zwave/cpp/src/command_classes/ThermostatSetpoint.cpp	2012-12-23 09:58:55.626496741 +0100
 @@ -180,7 +180,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
@@ -810,10 +828,10 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/UserCode.cpp open-zwave/cpp/src/command_classes/UserCode.cpp
---- open-zwave.orig/cpp/src/command_classes/UserCode.cpp	2012-12-19 17:09:52.170454351 +0100
-+++ open-zwave/cpp/src/command_classes/UserCode.cpp	2012-12-19 17:11:37.469632731 +0100
-@@ -136,7 +136,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/UserCode.cpp open-zwave/cpp/src/command_classes/UserCode.cpp
+--- open-zwave.orig/cpp/src/command_classes/UserCode.cpp	2012-12-23 09:54:43.517906266 +0100
++++ open-zwave/cpp/src/command_classes/UserCode.cpp	2012-12-23 09:58:55.630496560 +0100
+@@ -142,7 +142,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
  		msg->Append( UserNumberCmd_Get );
@@ -822,7 +840,7 @@
  		GetDriver()->SendMsg( msg, _queue );
  		return true;
  	}
-@@ -147,7 +147,7 @@
+@@ -153,7 +153,7 @@
  	msg->Append( GetCommandClassId() );
  	msg->Append( UserCodeCmd_Get );
  	msg->Append( _userCodeIdx );
@@ -831,7 +849,7 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-@@ -266,7 +266,7 @@
+@@ -268,7 +268,7 @@
  		{
  			msg->Append( s[i] );
  		}
@@ -840,10 +858,10 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  		return true;
  	}
-diff -ru open-zwave.orig/cpp/src/command_classes/Version.cpp open-zwave/cpp/src/command_classes/Version.cpp
---- open-zwave.orig/cpp/src/command_classes/Version.cpp	2012-12-19 17:09:52.314447756 +0100
-+++ open-zwave/cpp/src/command_classes/Version.cpp	2012-12-19 17:11:37.469632731 +0100
-@@ -108,7 +108,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/Version.cpp open-zwave/cpp/src/command_classes/Version.cpp
+--- open-zwave.orig/cpp/src/command_classes/Version.cpp	2012-12-23 09:54:43.525905903 +0100
++++ open-zwave/cpp/src/command_classes/Version.cpp	2012-12-23 09:58:55.630496560 +0100
+@@ -144,7 +144,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
  	msg->Append( VersionCmd_Get );
@@ -852,7 +870,7 @@
  	GetDriver()->SendMsg( msg, _queue );
  	return true;
  }
-@@ -193,7 +193,7 @@
+@@ -229,7 +229,7 @@
  			msg->Append( GetCommandClassId() );
  			msg->Append( VersionCmd_CommandClassGet );
  			msg->Append( _commandClass->GetCommandClassId() );
@@ -861,10 +879,10 @@
  			GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
  			return true;
  		}
-diff -ru open-zwave.orig/cpp/src/command_classes/WakeUp.cpp open-zwave/cpp/src/command_classes/WakeUp.cpp
---- open-zwave.orig/cpp/src/command_classes/WakeUp.cpp	2012-12-19 17:09:52.170454351 +0100
-+++ open-zwave/cpp/src/command_classes/WakeUp.cpp	2012-12-19 17:11:37.469632731 +0100
-@@ -151,7 +151,7 @@
+diff -u -r open-zwave.orig/cpp/src/command_classes/WakeUp.cpp open-zwave/cpp/src/command_classes/WakeUp.cpp
+--- open-zwave.orig/cpp/src/command_classes/WakeUp.cpp	2012-12-23 09:54:43.513906448 +0100
++++ open-zwave/cpp/src/command_classes/WakeUp.cpp	2012-12-23 09:58:55.634496379 +0100
+@@ -152,7 +152,7 @@
  	msg->Append( 2 );
  	msg->Append( GetCommandClassId() );
  	msg->Append( WakeUpCmd_IntervalGet );
@@ -873,7 +891,7 @@
  	GetDriver()->SendMsg( msg, _queue );
  
  	if( GetVersion() > 1 )
-@@ -161,7 +161,7 @@
+@@ -162,7 +162,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
  		msg->Append( WakeUpCmd_IntervalCapabilitiesGet );
@@ -882,16 +900,16 @@
  		GetDriver()->SendMsg( msg, _queue );
  	}
  
-@@ -285,7 +285,7 @@
+@@ -286,7 +286,7 @@
  		msg->Append( (uint8)(( interval >> 8 ) & 0xff) );	 
  		msg->Append( (uint8)( interval & 0xff ) );		
  		msg->Append( GetDriver()->GetNodeId() );
 -		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE );
 +		msg->Append( TRANSMIT_OPTION_ACK | TRANSMIT_OPTION_AUTO_ROUTE | TRANSMIT_OPTION_EXPLORE );
- 		GetDriver()->SendMsg( msg, Driver::MsgQueue_Send );
+ 		GetDriver()->SendMsg( msg, Driver::MsgQueue_WakeUp );
  		return true;
  	}
-@@ -408,7 +408,7 @@
+@@ -423,7 +423,7 @@
  		msg->Append( 2 );
  		msg->Append( GetCommandClassId() );
  		msg->Append( WakeUpCmd_NoMoreInformation );
@@ -900,4 +918,3 @@
  		GetDriver()->SendMsg( msg, Driver::MsgQueue_WakeUp );
  	}
  }
-Only in open-zwave/debian/patches: explore_routing.patch
