Index: open-zwave/cpp/src/Defs.h
===================================================================
--- open-zwave.orig/cpp/src/Defs.h	2012-12-19 16:54:28.040548361 +0100
+++ open-zwave/cpp/src/Defs.h	2012-12-19 16:56:09.163937788 +0100
@@ -90,7 +90,7 @@
 
 #define MAX_TRIES		3	// Retry sends up to 3 times
 #define MAX_MAX_TRIES		7	// Don't exceed this retry limit
-#define RETRY_TIMEOUT	2000		// Retry send after two seconds
+#define RETRY_TIMEOUT	40000		// Retry send after two seconds
 
 #define SOF												0x01
 #define ACK												0x06
