Index: open-zwave/cpp/build/linux/Makefile
===================================================================
--- open-zwave.orig/cpp/build/linux/Makefile	2012-07-18 15:44:49.460445594 +0200
+++ open-zwave/cpp/build/linux/Makefile	2012-07-18 16:25:26.435228587 +0200
@@ -98,4 +98,4 @@
 			$(patsubst %.cpp,%.o,$(vclasses)) \
 			$(patsubst %.cpp,%.o,$(pform)) \
 			$(patsubst %.cpp,%.o,$(indep)) vers.o
-	$(LD) -shared -Wl,-soname,libopenzwave.so -o $@ $+
+	$(LD) -shared -Wl,-soname,libopenzwave.so.1.0 -o $@.1.0 $+
